examples/whisper.py

```diff
@@ -214,30 +212,42 @@ def init_whisper(model_name="tiny.en"):
   enc = get_encoding(state['dims']['n_vocab'])
   return model, enc
 
-def transcribe_file(model, enc, filename):
-  waveform, sample_rate = librosa.load(filename, sr=RATE)
+def transcribe_waveform(waveform, model, enc):
+  n_frames = 3000
   log_spec = prep_audio(waveform)
-  lst = [enc._special_tokens["<|startoftranscript|>"], enc._special_tokens["<|notimestamps|>"]]
-  dat = model.encoder(Tensor(log_spec)).realize()
 
-  for i in range(50):
-    out = model.decoder(Tensor([lst]), dat).realize()
-    idx = int(out[0,-1].argmax().numpy().item())
-    lst.append(idx)
-    transcription = enc.decode(lst)
-    print(transcription)
-    if lst[-1] == enc._special_tokens["<|endoftext|>"]:
-      return  transcription
+  seek = 0
+  lst = [enc._special_tokens["<|startoftranscript|>"], enc._special_tokens["<|notimestamps|>"]]
+  final_transcription = ""
+  while seek < log_spec.shape[-1]:
+    log_spec_segment = log_spec[:, :, seek:seek+n_frames]
+    seek += n_frames
+
+    dat = model.encoder(Tensor(log_spec_segment)).realize()
+    #https://github.com/openai/whisper/blob/b38a1f20f4b23f3f3099af2c3e0ca95627276ddf/whisper/decoding.py#L524C70-L524C70
+    n_sample = model.decoder.positional_embedding.shape[0] // 2
+    for i in range(n_sample):
+      out = model.decoder(Tensor([lst]), dat).realize()
+      idx = int(out[0,-1].argmax().numpy().item())
+      lst.append(idx)
+      transcription = enc.decode(lst)
+      print(transcription)
+      if lst[-1] == enc._special_tokens["<|endoftext|>"]:
+        final_transcription += transcription
+        if seek == log_spec.shape[-1]:
+          return  final_transcription
+        lst = [enc._special_tokens["<|startoftranscript|>"], enc._special_tokens["<|notimestamps|>"]]
```

Comment:
Is this done in the reference impl?

Reply:
good point! for semplicity in the previous implementation I followed [this scenario](https://github.com/openai/whisper/blob/b38a1f20f4b23f3f3099af2c3e0ca95627276ddf/whisper/transcribe.py#L366).

In the current version, I have instead implemented the token feed